src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1366,12 +1399,16 @@ internal Collection<FunctionDescriptor> GetFunctionDescriptors(IEnumerable<Funct
                         }
                     }
 
-                    ValidateFunction(descriptor, httpFunctions);
-
                     if (descriptor != null)
                     {
+                        ValidateFunction(descriptor, httpFunctions);
                         functionDescriptors.Add(descriptor);
                     }
+                    else
+                    {
+                        string functionLanguage = _settingsManager.Configuration[ScriptConstants.FunctionWorkerRuntimeSettingName];
```

Comment:
This doesn't seem like the correct place for this message. The descriptor can be null for many other reasons, right?

Reply:
We are conditionally adding to _descriptorProviders [here](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L496) and throwing error only descriptor for a function is null.

Reply:
The contract above for `provider.TryCreate` is a bool, returning false and a null descriptor if the provider can't create for whatever reason (e.g. validation failures, no binding provider for a particular binding type, etc.) So it seems to me your error message would be misleading in such cases.